keps/sig-node/4205-psi-metric/README.md

```diff
@@ -0,0 +1,657 @@
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+- `k8s.io/kubernetes/pkg/kubelet/server/stats
+`: `2023-10-04` - ``
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Phase 1: Alpha
+
+- PSI integrated in kubelet behind a feature flag.
+- Unit tests to check the fields are populated in the 
+  Summary API response.
+
+#### Phase 2: Alpha
+
+- Implement Phase 2 of the enhancement which enables kubelet to 
+report node conditions based off PSI values.
+- Initial e2e tests completed and enabled if CRI implementation supports
+it.
+
+
+#### Beta
+
+- Feature gate is enabled by default.
+- Add documentation for the feature.
+- Extend e2e test coverage.
+- Allowing time for feedback.
+
+#### GA
+- TBD
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+### Upgrade / Downgrade Strategy
+
+NA
```

Comment:
Can you explain why upgrade/downgrade won't be a problem?

Reply:
For example, runc will be bumped to required 1.2.0 version so all the required components will be already at the expected levels, so there should be not problems during upgrade nor downgrades.